source/libNsdl/src/sn_nsdl.c

```diff
@@ -909,10 +909,14 @@ int8_t sn_nsdl_build_registration_body(struct nsdl_s *handle, sn_coap_hdr_s *mes
 
             *temp_ptr++ = '<';
             *temp_ptr++ = '/';
+            size_t path_len = 0;
+            if (resource_temp_ptr->static_resource_parameters->path) {
```

Comment:
Is this path != null check needed here? On what cases can it be NULL? If it was verified when item was added to list, this and few other places could be simplified a bit.